doc/sphinx/reference/onedim/grid-refinement.md

```diff
@@ -0,0 +1,228 @@
+0 and 1 that is specified by the user, $ x'_n $ is the derivative of the
+solution component being considered, and $ \displaystyle\max_j x'_n $ and
+$ \displaystyle\min_j x'_n $ are the maximum and minimum values of the
+solution component derivative across the domain. Points where $ C_j $ is greater
+than $ C_{max} $ are marked to have an additional grid point inserted to the
+right of them. Visually, the minimum and maximum global quantities are shown in the
+image below.
+
+```{figure} /_static/images/onedim/refinement_curve_metric_plot.svg
+:width: 75%
+:alt: Diagram of the curve refinement criterion.
+:align: center
+
+Diagram of the curve refinement criterion.
+```
+
+**Note:** The slope is computed using a backward difference scheme and so the estimate of
+the derivative at a grid point depends on the solution at the grid point and the
+solution at the grid point to the right.
+
+$$
+  x'_{n, j} \approx \frac{x_{n,j+1} - x_{n,j}}{z_{j+1} - z_{j}}
+$$
+
+
+### Prune
+
+This refinement option works in contrast to the other refinement options. Instead of
+adding grid points, this option removes grid points that are not needed. The pruning
+parameter helps the grid refinement algorithm identify areas where the grid may be
+overly refined. It is most useful when running multiple simulations where the boundary
+conditions change, and the location of the regions where grid refinement is needed
+changes. In this case, the prune parameter can be used to remove grid points that are
+no longer needed.
+
+You specify a pruning value $ P $ that serves as a lower-bound on
+the ratio quantities. This helps prevent unnecessary grid refinement by
+removing points that are not needed, optimizing computational efficiency without
+sacrificing the accuracy of the solution. The following criteria are used to decide
+if a point is marked for removal:
+
+$$
+  \frac{S_j}{S_{\text{max}}} < P \quad \text{or}
+  \quad \frac{C_j}{C_{\text{max}}} < P
+$$
+
+If $ \frac{S_j}{S_{max}} $, or $ \frac{C_j}{C_{max}} $ are less
+than $ P $, then the point is marked for removal.
+
+For example, consider a case where the slope refinement criteria is set to 0.5, we
+can look at the pruning criteria as $ S_j < S_{max} P $. In this form
+it is easy to see that the pruning parameter defines "What fraction of the original
+$ S_{max} $ is acceptable enough to remove a point?" Continuing with this example,
+consider a case where the refinement has progressed such that these are the values of
+the $ S_j $ at each point in a domain with 10 grid points.
+
+$$
+  S_j = [0.11, 0.16, 0.05, 0.1, 0.22, 0.09, 0.05, 0.02, 0.37, 0.29]
+$$
+
+All of the values are below the original $ S_{max} $ of 0.5, but some are much
+lower than that threshold. If the pruning parameter is set to a value of 0.2, then
+the effective cutoff threshold for removing a point would be 0.5*0.2 = 0.1. With this
+pruning criteria, the pruned data would be:
+
+$$
+  S_j = [0.11, 0.16, 0.1, 0.22, 0.37, 0.29]
+$$
```

Comment:
This example is interesting, but I think the end result you get isn't quite right -- the sum of $S_j$ should be a constant, since the change in the solution ends up getting absorbed into the neighboring points. This might be more clear if you define a corresponding array with values of $x_j$ before and after pruning.

Reply:
I'm thinking of something simpler: just marking the points that are below the pruning criteria in red.